daemon/src/test/java/org/mvndaemon/mvnd/cache/impl/CacheFactoryTest.java

```diff
@@ -64,6 +66,12 @@ public void assertPutGet(Path tempDir, final Cache<String, CacheRecord> cache, i
         Assertions.assertEquals(record2, cache.get(k2));
         Assertions.assertFalse(record2.invalidated);
 
+        /* Make sure the new content is written a couple of ms later than the original lastModifiedTime */
+        final long deadline = Files.readAttributes(file1, BasicFileAttributes.class).lastModifiedTime().toMillis() + 10;
+        while (System.currentTimeMillis() < deadline) {
+            /* Wait */
+        }
```

Comment:
This is my attempt to "fix" the `CacheFactoryTest.timestampCache()` failures on GH actions. They occur only on Linux, and my working hypothesis was that the new content was written to the file so fast after the original content that the timestamp did not change. It is not an issue on Mac, because `BasicFileAttributes.fileKey()` changes when the content is changed. On Linux BasicFileAttributes.fileKey() stays constant for the given path and thus we effectively depend only on the timestamp. I wonder we should come around that? This "fix" is hiding the problem rather than solving it. Do we, under these circumstances, still find `TimestampCacheFactory` good enough for Linux?

Reply:
I'd say that's fine.  I doubt this can really happen as that would mean an artifact is built while being consumed at the same time, which means that either the reactor dependencies are broken, or that the user is building the same project in two different concurrent maven invocation, which is bound to fail anyway, so I'm not too worried about that.

Reply:
Is this PR good to merge then?

Reply:
@ppalaga The loop should contain a `Thread.sleep(1)` to avoid cpu intensive looping. Otherwise good.